### PR TITLE
Add fixed stable version of controller

### LINKED ISSUE #11: Fix ocrd/all version
**Problem**
When building the Contoller image, the current "ocrd/all" is always downloaded. So we cannot provide a stable state of the controller in our stack with this agile base. In addition, it is a hindrance to development and demos if the long download is triggered when the entire stack is built and the controller may no longer function afterwards.

**Solution**
The version of ocrd/all should be entered as a fixed value in the build script of the controller and passed as an argument. The current version of ocrd/all can also be used if for example "latest" is set as argument

### PR DESCRIPTION
**Pull request by markusweigelt**
Fixes #11

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -34,7 +35,7 @@ jobs:
         date -u +"build_date=%Y-%m-%dT%H:%M:%SZ" >> $GITHUB_ENV
         echo "vcs_ref=`git rev-parse --short HEAD`" >> $GITHUB_ENV
```
Comment (bertsky) [inline .github/workflows/publish.yml:37]:
```suggestion
echo "vcs_tag=`git describe --tags --abbrev=0 2>/dev/null`" >> $GITHUB_ENV

```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -44,5 +45,17 @@ jobs:
...
-        cache-to: type=gha,mode=max  
-    
+        cache-to: type=gha,mode=max
+
+    - name: Build the stable OCR-D Controller image and deploy to GitHub Container Repository
+      uses: docker/build-push-action@v4
+      with:
+        context: .
+        push: true
+        tags: ${{ env.REGISTRY_PATH }}/ocrd_controller:stable
```
Comment (bertsky) [inline .github/workflows/publish.yml:55]: Perhaps we can even add some release tag here:
```suggestion
        tags: ${{ env.REGISTRY_PATH }}/ocrd_controller:stable,${{ env.REGISTRY_PATH }}/ocrd_controller:${{ env.vcs_tag }}
```

(Will be just `HEAD` if there is no tag for it.)

**Review by bertsky — APPROVED** [review]
Great stuff. So we can manually trigger the action by editing `STABLE_VERSION` (getting a new `latest` and `stable`) or automatically by any other change (getting a new `latest`, because the other is cached). Right?

**Comment by markusweigelt** [comment]
> Great stuff. So we can manually trigger the action by editing `STABLE_VERSION` (getting a new `latest` and `stable`) or automatically by any other change (getting a new `latest`, because the other is cached). Right?

Atm the stable version rebuilds with same behaviour like the latest version but with the different of using the parent version. So when something in Controller repo is changed than the stable version rebuilds too.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (markusweigelt) [reply .github/workflows/publish.yml:55]: The question was whether saving the tags will cause problems on GitHub. I assumed that the stable tag simply overrides the stable image version.

Comment (bertsky) [reply .github/workflows/publish.yml:55]: > I assumed that the stable tag simply overrides the stable image version.

I'm sure it does. This is just to make sure we **also** produce some persistent images ourselves from time to time (viz. whenever we decide to make a release).

Comment (bertsky) [reply .github/workflows/publish.yml:55]: Why didn't you commit my suggestions?

Comment (markusweigelt) [reply .github/workflows/publish.yml:55]: oh I thought I had already done that. Was not intentional. I'll change that directly.

Comment (markusweigelt) [reply .github/workflows/publish.yml:55]: Added your suggest changes with seperate pr #22

**Pull request merged**